### ISSUE TITLE
[Flang] Irritating warning about NUMERIC_STORAGE_SIZE while not used

### ISSUE THREAD

**Issue by mjklemm** [llvm/llvm-project#100376]
In the following code, `NUMERIC_STORAGE_SIZE` is not being used (and technically not visible):

```Fortran
program reproducer
    use iso_fortran_env, only: output_unit
    implicit none
end program reproducer
```

Flang still a warning, when compiling the file like this:

```
> flang-new -fdefault-real-8 -c reproducer.f90
reproducer.f90:2:9: warning: NUMERIC_STORAGE_SIZE from ISO_FORTRAN_ENV is not well-defined when default INTEGER and REAL are not consistent due to compiler options
      use iso_fortran_env, only: output_unit
          ^^^^^^^^^^^^^^^
```

Should this warning be issued when `NUMERIC_STORAGE_SIZE` is not visible?

**Comment by mjklemm** [comment]
@klausler @kiranchandramohan @clementval @tblah Can you please comment?  I think this maybe a bug.  Is it?

**Comment by mjklemm** [comment]
I'd like to plea not guilty, as the warning still appears with commit 22eb290a9696 which is the one before I merged the ISO_FORTRAN_ENV stuff.  When looking at the code it seems to be a static warning that only depends on the sizes of of INTEGER and REAL depending on the compiler options picked.

**Comment by klausler** [comment]
Can confirm, it shows up earlier.
